solutions/community-deploy/communitybuild.sh

```diff
@@ -0,0 +1,133 @@
+                echo "Docker is installed and $(docker --version)."
+        else
+                echo "Docker not installed"
+                printf "\n################################################################################\n"
+                printf "\tInstalling Docker Engine\n"
+                printf "\n################################################################################\n"
+                yum install -y yum-utils
+                yum-config-manager \
+    --add-repo \
+    https://download.docker.com/linux/centos/docker-ce.repo -y
+                yum install -y docker-ce docker-ce-cli containerd.io docker-compose-plugin
+                systemctl start docker
+                docker --version
+#               exit 1
+        fi
+}
+DockerCheck
+
+function DockerComposeCheck() {
+        docker-compose --version >/dev/null 2>&1
+        if [ $? -eq 0 ]; then
+                echo "Docker-Compose is installed and $(docker-compose --version)."
+        else
+                echo "Docker-Compose not installed"
+                printf "\n################################################################################\n"
+                printf "\tInstalling Docker-Compose\n"
+                printf "\n################################################################################\n"
+                curl -SL https://github.com/docker/compose/releases/download/v2.5.0/docker-compose-linux-x86_64 -o /usr/local/bin/docker-compose
+                chmod +x /usr/local/bin/docker-compose
+                ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose
+                docker-compose --version
+#               exit 1
+        fi
+}
+DockerComposeCheck
+
+# Compile and Build CORTX Stack
+docker pull ghcr.io/seagate/cortx-build:rockylinux-8.4
+
+# Clone the CORTX repository
+cd /mnt && git clone https://github.com/Seagate/cortx --recursive --depth=1
+
+# Checkout main branch for generating CORTX packages
+docker run --rm -v /mnt/cortx:/cortx-workspace ghcr.io/seagate/cortx-build:rockylinux-8.4 make checkout BRANCH=$BRANCH
+
+# Validate CORTX component clone status
+cd /mnt/cortx/ 
+for component in cortx-motr cortx-hare cortx-rgw-integration cortx-manager cortx-utils cortx-ha cortx-rgw
+do 
+echo -e "\n==[ Checking Git Branch for $component ]=="
+pushd $component
+git status | egrep -iw 'Head|modified|Untracked'
+    if [ $? -eq 0 ]; then
+        echo "GIT STATUS PENDING"
+        exit 1
+    else    
+        popd
+    fi    
+done && cd -
+
+# Build the CORTX packages
+docker run --rm -v /var/artifacts:/var/artifacts -v /mnt/cortx:/cortx-workspace ghcr.io/seagate/cortx-build:rockylinux-8.4 make clean cortx-all-rockylinux-image
+
+function PacketValidation() {
+        ls -l /var/artifacts/0 | egrep -iw '3rd_party|python_deps'
+        if [ $? -eq 0 ]; then
+            echo "Required Packages Are Available"
+        else
+            echo "Required Packages Are Not Available To Proceed Further"
```

Comment:
keep only first letter of first word capital

Reply:
Done